modules/openapi-generator/src/main/java/org/openapitools/codegen/ComplexType.java

```diff
@@ -0,0 +1,61 @@
+package org.openapitools.codegen;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public interface ComplexType {
+
+    /**
+      * @return basic type - no generics supported.
+     */
+    String getBaseType();
+
+    /**
+     * @return complex type that can contain type parameters - like {@code List<Items>} for Java
+     */
+    String getComplexType();
+
+    /**
+     * @return true if set, map or list.
+     */
+    boolean isContainer();
+
+    /**
+     * @return the composed schemas that might make up this type, null otherwise.
+     */
+    CodegenComposedSchemas getComposedSchemas();
+
+    /**
+     * @return the inner type in the case of a type with type parameters
+     */
+    ComplexType getInner();
+
+    /**
+     * Recursively collect all necessary imports to include so that the type may be resolved.
+     *
+     * @param includeContainerTypes whether or not to include the container types in the returned imports.
+     * @return all of the imports
+     */
+    default Set<String> getImports(boolean includeContainerTypes) {
+        Set<String> imports = new HashSet<>();
+            if (includeContainerTypes || !this.isContainer()) {
+                if (this.getComposedSchemas() != null) {
+                    CodegenComposedSchemas composed = (CodegenComposedSchemas) this.getComposedSchemas();
+                    List<CodegenProperty> allOfs =  composed.getAllOf() == null ? Collections.emptyList() : composed.getAllOf();
+                    List<CodegenProperty> oneOfs =  composed.getOneOf() == null ? Collections.emptyList() : composed.getOneOf();
+                    Stream<CodegenProperty> innerTypes = Stream.concat(allOfs.stream(), oneOfs.stream());
+                    innerTypes.flatMap(cp -> cp.getImports(includeContainerTypes).stream()).forEach(s -> imports.add(s));
+                } else {
+                    imports.add(this.getComplexType());
+                    imports.add(this.getBaseType());
+                }
+            }
+        if (this.getInner() !=null) {
+            imports.addAll(this.getInner().getImports(includeContainerTypes));
+        }
+        return imports;
+    }
+}
```

Comment:
Can you merge this interface into IJsonSchemaValidationProperties?
- How about using isMap and isArray to determine if one needs to look in the object to find more imports?
- Also anyType objects can define items, properties, and additionalProperties so those imports should also be pulled in. I don't think your code handles that use case yet.

Reply:
Why would getImports not work for an AnyType? I made all the other changes, I just wasn't sure about why this code wouldn't work fine if `this` is an any type? It will just recurse in to items, composedSchemas and additionalProperties without discrimination, right?

Reply:
You are correct that if we cover all of the base cases with sequential if statements that add imports then we don't need special AnyType handling